local.go

```diff
@@ -196,46 +197,40 @@ func getDefaultMethod(funcInfo *FunctionInfo) Method {
 // Combine location and stats because maybe in the future we'll add evaluation times
 // to functionStats?
 func getStartingLocation(f Function, funcs functions, funcInfo *FunctionInfo, initX []float64, stats *Stats, settings *Settings) (Location, error) {
-	var l Location
-
-	l.X = make([]float64, len(initX))
-	copy(l.X, initX)
-	if funcInfo.IsGradient || funcInfo.IsFunctionGradient {
-		l.Gradient = make([]float64, len(l.X))
+	dim := len(initX)
+	loc := Location{
```

Comment:
Would you mind changing the various "loc" back to their original name? I'm okay with fixing them, but it obscures this diff quite a bit. We can fix it in a future PR.